core/rfb.js

```diff
@@ -804,6 +808,15 @@ export default class RFB extends EventTargetMixin {
                    size.w + 'x' + size.h);
     }
 
+    _requestKeyFrame() {
+        if (this._enabledContinuousUpdates) {
+            RFB.messages.fbUpdateRequest(this._sock, false, 0, 0,
+                                         this._fbWidth, this._fbHeight);
+        } else {
+            this._keyFrameNeeded = true;
+        }
+    }
```

Comment:
This seems overly complex to work around a bug in Firefox as the feature isn't deemed ready for usage there yet.

And I don't understand why this is needed. You wrote you worked around it by setting a frame duration?

Reply:
I see. In that case, I'm not sure if it's worth the complexity. Can we leave it out for now so we can focus on getting the core stuff merged?

Reply:
We can leave it out, but how should we handle decoding failures then? Raise an exception?

Reply:
That seems reasonable. That's how we handle problems in the other decoders.

VNC is a reliable transport, so it should not expect dropped frames or such that other video transports might have to deal with.

Reply:
> VNC is a reliable transport, so it should not expect dropped frames or such that other video transports might have to deal with.

Yes, but it is apparent that the browser may choose to drop the decoding context based on other things, which leads to decoding failures when no key frame is received after such an event.

Currently, in Firefox, this happens as a result of a bug, but I don't see any guarantees in the API that this won't happen under other circumstances. Maybe they'll decide at some point that decoding when the window is off-screen is useless, so they start dropping packets to the decoder?

Reply:
I think that sounds unreasonable for the browser. It doesn't know what we intend to do with the data. If it isn't explicitly specified that it can drop data if it feels like it, then we should assume it will do what is requested.